go_tests/bal_build_test/bal_build_test.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"testing"
+
+	cp "github.com/otiai10/copy"
+)
+
+func setup(dir string) (string, error) {
+	tmpDir, err := os.MkdirTemp("", "run-test-resources")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temporary directory: %v", err)
+	}
+
+	err = cp.Copy(dir, tmpDir)
+	return tmpDir, nil
+}
+
+func TestBuildProjectByGivingTargertDir(t *testing.T) {
+	testResourcesDir := "/home/pabadhi/myGoProjects/ballerina-lang/cli/ballerina-cli/src/test/resources/test-resources/validApplicationProject"
+	tmpDir, err := setup(testResourcesDir)
+	defer os.RemoveAll(tmpDir)
+	projectPath := filepath.Join(tmpDir)
+	testTargetDir := filepath.Join(tmpDir, "test_target")
+	err = os.Mkdir(testTargetDir, os.ModePerm)
+	cmd := exec.Command("/home/pabadhi/myGoProjects/ballerina-lang/distribution/zip/jballerina-tools/build/extracted-distributions/jballerina-tools-2201.9.0-SNAPSHOT/bin/bal_linux_amd64", "build", "--target-dir", testTargetDir, projectPath)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatalf("Failed to build Ballerina project: %v\nOutput: %s", err, output)
+	}
+	expectedFiles := []string{
+		filepath.Join(testTargetDir, "bin", "winery.jar"),
+		filepath.Join(testTargetDir, "cache", "foo", "winery", "0.1.0", "java17", "foo-winery-0.1.0.jar"),
+	}
+
+	for _, file := range expectedFiles {
+		if _, err := os.Stat(file); os.IsNotExist(err) {
+			t.Errorf("Expected file does not exist: %s", file)
+		}
+	}
+
+}
+
+func TestBuildBalFile(t *testing.T) {
+	testResourcesDir := "/home/pabadhi/myGoProjects/ballerina-lang/cli/ballerina-cli/src/test/resources/test-resources/valid-bal-file"
+	tmpDir, err := setup(testResourcesDir)
+	validBalFilePath := filepath.Join(tmpDir, "hello_world.bal")
+	defer os.RemoveAll(tmpDir)
+	err = os.Chdir(filepath.Dir(validBalFilePath))
+	if err != nil {
+		t.Fatalf("Failed to change directory: %v", err)
+	}
+	cmd := exec.Command("/home/pabadhi/myGoProjects/ballerina-lang/distribution/zip/jballerina-tools/build/extracted-distributions/jballerina-tools-2201.9.0-SNAPSHOT/bin/bal_linux_amd64", "build", validBalFilePath)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatalf("Failed to build Ballerina file: %v\n%s", err, output)
+	}
+	fmt.Println(validBalFilePath)
+	expectedOutputPath := "/home/pabadhi/myGoProjects/ballerina-lang/go_tests/bal_build_test/bal_build_outputs/build-hello-world-bal.txt"
```

Comment:
Instead of having hardcoded paths, you can derive the paths to the resources using relative paths to the current location of the tests